iroh-p2p/src/service.rs

```diff
@@ -313,13 +313,25 @@ impl<KeyStorage: Storage> Libp2pService<KeyStorage> {
                     if protocols
                         .iter()
                         .any(|p| p.as_bytes() == kad::protocol::DEFAULT_PROTO_NAME)
+                    {
+                        for addr in &listen_addrs {
+                            if let Some(kad) = self.swarm.behaviour_mut().kad.as_mut() {
+                                kad.add_address(&peer_id, addr.clone());
+                            }
+                        }
+                    }
+
+                    // Inform autonat about identified peers
+                    // TODO: expose protocol name on `libp2p::autonat`.
+                    // TODO: should we remove them at some point?
+                    if protocols
+                        .iter()
+                        .any(|p| p.as_bytes() == b"/libp2p/autonat/1.0.0")
```

Comment:
How about exposing the protocol ID as a constant from `libp2p-autonat` and using that here instead?

https://github.com/libp2p/rust-libp2p/blob/ea487aebfe6eb672b05d2bec2d9d79bbd92450ba/protocols/autonat/src/protocol.rs#L34

Reply:
yep, that's the todo above, I can make a PR

Reply:
https://github.com/libp2p/rust-libp2p/pull/2717